File: src/main/java/com/pawan/choure/java/handler/TradeHandlerSimple.java
package com.pawan.choure.java.handler;

import org.springframework.messaging.Message;
import org.springframework.messaging.MessageHandler;
import org.springframework.messaging.MessagingException;

public abstract class TradeHandlerSimple implements MessageHandler {
    @Override
    public void handleMessage(Message<?> message) throws MessagingException {
        System.out.println("TradeHandlerSimple Header" + message.getHeaders());
        System.out.println("TradeHandlerSimple PayLoad Hello" +message.getPayload());
        System.out.println("TradeHandlerSimple MessageToString"+message.toString());
        receiveAndAcknowledge(message.getPayload());
    }
    protected abstract void receiveAndAcknowledge(Object payload);
}

File: src/main/kotlin/com/pawan/choure/kotlin/handler/TradeHandlerSimple.kt
package com.pawan.choure.kotlin.handler

import org.springframework.messaging.Message
import org.springframework.messaging.MessageHandler
import org.springframework.messaging.MessagingException

abstract class TradeHandlerSimple : MessageHandler {

    @Throws(MessagingException::class)
    open override fun handleMessage(message: Message<*>) {
        println("TradeHandlerSimple Header" + message.headers)
        println("TradeHandlerSimple PayLoad Hello" +message.payload)
        println("TradeHandlerSimple MessageToString$message")
        val payload = message.payload
        receiveAndAcknowledge(payload)
    }

    protected abstract fun receiveAndAcknowledge(status: Any)
}

